build.go

```diff
@@ -255,6 +257,37 @@ func isBuildpackID(bp string) bool {
 	return false
 }
 
+func ensureBPSupport(bpPath string) (err error) {
```

Comment:
I feel like this type of logic belongs in the blob package. The idea was to encapsulate all logic about data formats in that package.

For now it looks like directory blobs are never allowed on windows (b/c of the perm issues). If we wanted to support directory blobs for an app dir or adding a buildpack at build time we could create strict and non-strict blob constructor options to that package.

Reply:
Deferred to #286